### PR TITLE
Fix defects found by am_policies_test under WinQt

### PR DESCRIPTION
**Pull request by vlantonov**
Fixed Unit test output.
Refactored PolicyHandlerTest.
Fixed DataAccessor expectation release.

Related issue: [APPLINK-27158](https://adc.luxoft.com/jira/browse/APPLINK-27158)

### CODE REVIEW

**Comment by vlantonov** [comment]
Please review: @AByzhynar, @dcherniev, @OHerasym, @VProdanov, @wolfylambova22, @pvvasilev

File: src/components/application_manager/test/policies/policy_handler_test.cc [hunk 1]
```diff
@@ -96,18 +97,19 @@ class PolicyHandlerTest : public ::testing::Test {
...
   utils::SharedPtr<application_manager_test::MockApplication> mock_app_;
-  connection_handler_test::MockConnectionHandler conn_handler;
-  protocol_handler_test::MockSessionObserver mock_session_observer;
-  application_manager_test::MockStateController mock_state_controller;
+  connection_handler_test::MockConnectionHandler conn_handler_;
+  protocol_handler_test::MockSessionObserver mock_session_observer_;
+  application_manager_test::MockStateController mock_state_controller_;
   PolicyHandler policy_handler_;
   utils::SharedPtr<policy_manager_test::MockPolicyManager> mock_policy_manager_;
   application_manager_test::MockApplicationManager app_manager_;
```
Comment (AByzhynar) [inline src/components/application_manager/test/policies/policy_handler_test.cc]: @vlantonov The same here. Add prefix `mock_` please

File: src/components/application_manager/test/policies/policy_handler_test.cc [hunk 2]
```diff
@@ -1019,7 +1026,8 @@ TEST_F(PolicyHandlerTest, SendOnAppPermissionsChanged) {
   // Check expectations
   EXPECT_CALL(app_manager_, application_by_policy_id(kPolicyAppId_))
       .WillOnce(Return(application));
-  EXPECT_CALL(*MockMessageHelper::message_helper_mock(),
+  EXPECT_CALL(*application, app_id()).WillOnce(Return(0u));
```
Comment (AByzhynar) [inline src/components/application_manager/test/policies/policy_handler_test.cc]: @vlantonov Is this a real application class object? If no - please add prefix `mock_` to variable name

File: src/components/application_manager/test/policies/policy_handler_test.cc [hunk 3]
```diff
@@ -120,15 +122,17 @@ class PolicyHandlerTest : public ::testing::Test {
   std::string default_hmi_;
   const std::string kPreloadPTFile_;
   const std::string kAppStorageFolder_;
-  ApplicationSet test_app;
-  sync_primitives::Lock app_lock;
-  DataAccessor<ApplicationSet> app_set;
+  ApplicationSet test_app_;
```
Comment (AByzhynar) [inline src/components/application_manager/test/policies/policy_handler_test.cc]: @vlantonov Should be renamed as it's not just one app but the entire set of apps.

File: src/components/application_manager/test/policies/policy_handler_test.cc [hunk 4]
```diff
@@ -96,18 +97,19 @@ class PolicyHandlerTest : public ::testing::Test {
...
+      , mock_message_helper_(MockMessageHelper::message_helper_mock()) {}
 
  protected:
   NiceMock<MockPolicySettings> policy_settings_;
   NiceMock<event_engine_test::MockEventDispatcher> mock_event_dispatcher_;
   utils::SharedPtr<application_manager_test::MockApplication> mock_app_;
-  connection_handler_test::MockConnectionHandler conn_handler;
-  protocol_handler_test::MockSessionObserver mock_session_observer;
-  application_manager_test::MockStateController mock_state_controller;
+  connection_handler_test::MockConnectionHandler conn_handler_;
```
Comment (vlantonov) [inline src/components/application_manager/test/policies/policy_handler_test.cc]: @AByzhynar Fixed in 35d17348a32a8b2eff7c1f66add80ca38ccf8eec

File: src/components/application_manager/test/policies/policy_handler_test.cc (hunk 1 above)
Comment (vlantonov) [inline src/components/application_manager/test/policies/policy_handler_test.cc]: @AByzhynar Fixed in 35d17348a32a8b2eff7c1f66add80ca38ccf8eec

File: src/components/application_manager/test/policies/policy_handler_test.cc (hunk 3 above)
Comment (vlantonov) [inline src/components/application_manager/test/policies/policy_handler_test.cc]: @AByzhynar Fixed in 35d17348a32a8b2eff7c1f66add80ca38ccf8eec

File: src/components/application_manager/test/policies/policy_handler_test.cc (hunk 2 above)
Comment (vlantonov) [inline src/components/application_manager/test/policies/policy_handler_test.cc]: @AByzhynar Fixed in 35d17348a32a8b2eff7c1f66add80ca38ccf8eec

**Pull request merged**